src/rust/engine/fs/src/store.rs

```diff
@@ -680,7 +700,7 @@ impl Store {
     destination: PathBuf,
     digest: Digest,
     is_executable: bool,
-  ) -> BoxFuture<LoadMetadata, String> {
+  ) -> BoxFuture<(std::fs::File, LoadMetadata), String> {
```

Comment:
Although this is a private method, it still feels like it would be good to document the fact that the file will not be `fsync`ed at the end of the call.